### PR TITLE
Support `cpu.frequency` hardware requirement for ``mrack``

### PR DESCRIPTION
**Pull request by skycastlelily**
Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] extend the test coverage
* [x] update the specification
* [x] mention the version
* [x] include a release note

### CODE REVIEW

**Comment by happz** [comment]
Blocked on https://github.com/teemtee/tmt/pull/3295

File: spec/hardware/cpu.fmf [hunk 1]
```diff
@@ -70,7 +70,7 @@ description: |
     are also useful resources.
 
     .. versionchanged:: 1.38
-       ``beaker`` plugins supports ``stepping``
+       ``beaker`` plugins supports ``frequency`` and ``stepping``
```
Comment (happz) [inline spec/hardware/cpu.fmf:73]: Both release note and this `versionchanged` need an update, it's no longer 1.38 but 1.39.

**Comment by skycastlelily** [comment]
Rebased and updated,and it just occurred to me that I should update first
in order to put the code addressing the comments in one commit, then
rebased, will keep that in mind^^

**Comment by skycastlelily** [comment]
Rebased and updated,and it just occurred to me that I should update first in order to put the code addressing the comments in one commit, then rebased, will keep that in mind^^

File: spec/hardware/cpu.fmf (hunk 1 above)
Comment (thrix) [reply spec/hardware/cpu.fmf]: addressed

File: spec/hardware/cpu.fmf [hunk 2]
```diff
@@ -115,15 +118,20 @@ example:
     cpu:
         hyper-threading: true
 
+  - |
+    # Request a CPU with specified frequency.
```
Comment (martinhoyer) [inline spec/hardware/cpu.fmf:122]: There is already the frequency example above, stating MHz is default and that it accepts different units, maybe best to not add another example?
https://tmt.readthedocs.io/en/stable/spec/hardware.html#cpu

File: tmt/hardware.py [hunk 3]
```diff
@@ -737,7 +737,7 @@ def _cast_int(raw_value: Any) -> int:
             )
 
 
-class NumberConstraint(Constraint[float]):
+class NumberConstraint(Constraint['Size']):
```
Comment (martinhoyer) [inline tmt/hardware.py:740]: hmm, I might be missing something. Why this change?

Comment (happz) [reply tmt/hardware.py:740]: Doesn't look correct indeed.

File: spec/hardware/cpu.fmf (hunk 2 above)
Comment (skycastlelily) [reply spec/hardware/cpu.fmf]: Updated,thanks for your review:)

File: tmt/hardware.py (hunk 3 above)
Comment (happz) [reply tmt/hardware.py:740]: Same here ;)

Comment (skycastlelily) [reply tmt/hardware.py]: update:https://github.com/teemtee/tmt/pull/3297/commits/7df0e709e1addbd4ebc62955511e43de37fd341a.

Sorry,mentor, I somehow overlooked the later part of the sentence :sweat_smile:

File: spec/hardware/cpu.fmf [hunk 4]
```diff
@@ -125,8 +125,8 @@ example:
 
 link:
   - implemented-by: /tmt/steps/provision/artemis.py
-    note: "``cpu.vendor``, ``cpu.vendor-name`` and ``cpu.hyper-threading`` not implemented yet"
+    note: "``cpu.vendor``, ``cpu.vendor-name``, ``cpu.frequency`` and ``cpu.hyper-threading`` not implemented yet"
   - implemented-by: /tmt/steps/provision/mrack.py
-    note: "``cpu.flag``, ``cpu.processors``, ``cpu.model``, ``cpu.cores``, ``cpu.model-name``, ``cpu.hyper-threading``, ``cpu.stepping``, ``cpu.family`` and ``cpu.vendor-name`` only"
+    note: "``cpu.sockets``, ``cpu.threads``, ``cpu.cores-per-socket``, ``cpu.threads-per-core`` not implemented yet"
```
Comment (therazix) [inline spec/hardware/cpu.fmf:130]: Shouldn't `family-name` and `vendor` also be mentioned here?
```suggestion
    note: "``cpu.sockets``, ``cpu.threads``, ``cpu.cores-per-socket``, ``cpu.threads-per-core``, ``cpu.family-name``, ``cpu.vendor`` not implemented yet"
```

Comment (skycastlelily) [reply spec/hardware/cpu.fmf]: Update:https://github.com/teemtee/tmt/pull/3297/commits/2a293b53b86df42771ce5bc6eaffdf1a8f356bb4
Thanks for your review:)

**Review by psss — APPROVED** [review]
Looks good, thanks!

**Comment by psss** [comment]
Red jobs are irrelevant failures, merged.

**Comment by skycastlelily** [comment]
:tada:

**Pull request merged**